Rejig introduction

aanand opened the pull request:
Resolves #944.

- Clearer overview in the first paragraph
- Emphasise development environments
- Mention that production is possible but not (yet) recommended, point to roadmap
- Tighten up the introductory example

@fredlf don't merge yet - awaiting feedback from @bfirsh.

fredlf commented on docs/index.md:
```diff
@@ -5,60 +5,49 @@ page_keywords: documentation, docs,  docker, compose, orchestration, containers
-With Compose, you can build clusters of containers which provide the resources
-(services, volumes, etc.) needed to build and run a complete distributed
-application.
+Compose is a tool for defining and running complex applications with Docker. It
```
Vague antecedent of pronoun. Change "It" to "Compose".

fredlf commented on docs/index.md:
```diff
@@ -5,60 +5,49 @@ page_keywords: documentation, docs,  docker, compose, orchestration, containers
-application.
+Compose is a tool for defining and running complex applications with Docker. It
+lets you define a multi-container application in a single file, then spin it up
+in a single command which does everything that needs to be done to get it
```
Ditto, "it"-->"the app"

Also, can simplify "needs to be done" to "needed", so "...a single command that does everything needed to get the app runing."

fredlf commented on docs/index.md:
```diff
@@ -5,60 +5,49 @@ page_keywords: documentation, docs,  docker, compose, orchestration, containers
+We're actively working on making it better in that regard - see the
+[roadmap](https://github.com/docker/fig/blob/master/ROADMAP.md).
 
-Using Compose is basically a three-step process.
```
Letting readers know how many steps are coming before giving them the steps is really important and effective. It gives them a conceptual framework they can use to properly contextualize what comes next. Please don't cut it.

aanand replied:
Updated. It's actually a two-step process (the Dockerfile step is not fundamental to it, which was the motivation for trying to simplify).

fredlf commented on docs/index.md:
```diff
@@ -5,60 +5,49 @@ page_keywords: documentation, docs,  docker, compose, orchestration, containers
 
-(No more installing Postgres on your laptop!)
-
-Lastly, run `docker-compose up` and Compose will start and run your entire app.
```
Again, the "lastly" helps the reader know this is the final step, provides useful context for interpreting the good bits at the end. We need to leave these context cues in place, please.

aanand replied:
Fixed. This means we've now got a "Firstly" and a "Lastly" with nothing in between - is that OK?

fredlf replied:
That's fine, but since we've now told the reader a two-step process is coming, we could just number the steps, eh?

aanand replied:
Sensible. Done.

fredlf commented on docs/index.md:
```diff
@@ -5,60 +5,49 @@ page_keywords: documentation, docs,  docker, compose, orchestration, containers
-on Compose. It assumes a little knowledge of Python, but the concepts
-demonstrated here should be understandable even if you're not familiar with
-Python.
+Let's get a simple Python web app running on Compose. It assumes a little
```
Why are we getting a simply Python web app running now? Necessity? Whim? Curiosity? The reader needs context (e.g., "Let's get started" or "Let's see how this works by...", etc.) to know why we are doing what we are doing, when.

fredlf commented:
Fine with the changes, content-wise, but you took out a lot of the semantic navigational/conceptual cues that make things much easier for readers to correctly, confidently, and easily interpret what you're writing. See notes inline.

aanand commented:
Just saw. Thanks for the pointers - I'd assumed it was my work I was stomping on, not yours. Will revisit.

fredlf commented:
Heh, I wouldn't say stomping, just a natural part of the writing process. Our brains often create thoughts in an order that isn't always the most helpful for readers. That's why we have revision (re-vision, seeing again)!

bfirsh commented on docs/index.md:
```diff
@@ -5,52 +5,42 @@ page_keywords: documentation, docs,  docker, compose, orchestration, containers
-remote server, including cloud-based instances - anywhere a Docker daemon can
-run. Its primary use case is for development environments, but it can be used
-just as easily for staging or CI.
+Compose is especially good for development environments, staging servers and CI.
```
"Compose is for running development envionments ..."

bfirsh replied:
If this is misleading in context of the first paragraph, perhaps we should clarify this earlier on.

bfirsh commented on docs/index.md:
```diff
@@ -5,52 +5,42 @@ page_keywords: documentation, docs,  docker, compose, orchestration, containers
+Compose is especially good for development environments, staging servers and CI.
+It can be used in production, but it isn't (yet) recommended that you do so.
+We're actively working on making it better in that regard - see the
+[roadmap](https://github.com/docker/fig/blob/master/ROADMAP.md).
```
I think we need a separate section in the documentation to describe this in more detail, as explained in #944. We shouldn't land users on the roadmap in the second paragraph because that is only going to confuse them (they don't know what Compose really is yet).

fredlf replied:
That seems reasonable, but is what we have now sufficient for beta release, MVD?

bfirsh commented:
Much better, thanks! Comments inline.

bfirsh commented on docs/index.md:
```diff
@@ -5,52 +5,42 @@ page_keywords: documentation, docs,  docker, compose, orchestration, containers
-Next, you define the services that make up your app in `docker-compose.yml` so
-they can be run together in an isolated environment:
+Using Compose is a two-step process. First, define the services that make up
+your app in `docker-compose.yml`, using standard Docker configuration options:
```
Where's the Dockerfile gone? We're still describing development environments, and it's an essential part of the process of doing that.

fredlf replied:
@aanand felt it was not really part of Compose per se and removed it in an effort to simplify. I feel like it provides useful context, but maybe isn't necessary depending on the Docker knowledge we expect Compose users to have. I can't really speak to that.

bfirsh replied:
I like how it explains how you go from from code --> image --> complete running app

aanand replied:
I feel its value has diminished since we put it in (back when we were still explaining what Docker was).

Let's be explicit: how much Docker knowledge do we expect people landing on this page to have?

bfirsh replied:
This will be the landing page for Docker Compose, so possibly none.

Imagine somebody who wants to run a development environment, and they're told to try out Docker Compose. This will be the page they land on.

aanand replied:
OK. In that case, before I make any more revisions, can we be absolutely clear about the primary audiences for this page? I now understand those to include:

- People who know what Docker is, and want to know about Docker's solutions for container orchestration, but aren't necessarily interested in development environments
- People who don't know anything about Docker, but are investigating Compose as a solution for development environments

Anything to add to that list?

aanand commented:
OK, one more pass:

- Made stance on production clearer
- Reverted changes to the example (brought the Dockerfile back)
- Updated README.md to bring it in line

fredlf commented on README.md:
```diff
@@ -3,40 +3,48 @@ Docker Compose
+be done to get it running.
 
-Define your app's environment with Docker so it can be reproduced anywhere:
+Compose is great for development environments, staging servers and CI. We don't
```
Oxford comma needed after "staging servers" please. (Here and below)

fredlf commented:
LGTM. Nice job! Not sure why it's failing tests. Needs a rebase?

bfirsh commented on README.md:
````diff
@@ -3,40 +3,48 @@ Docker Compose
 ```yaml
 web:
   build: .
+  command: python app.py
````
Why's this moved?

Starefossen commented:
I still think there should be a mention in the README that this project was previously known as fig. It seams like the rebranding is creating some confusion.

aanand commented:
Updated `Dockerfile`/`docker-compose.yml` to reflect best practices. Also made use of Dockerfile syntax highlighting.

fredlf commented:
LGTM from me. @bfirsh are we good to merge?

bfirsh commented:
Woot. Looks great! Thanks @aanand and @fredlf.

bfirsh commented:
LGTM
